packages/ra-core/src/controller/useListParams.ts

```diff
@@ -163,21 +164,31 @@ const useListParams = ({
     }, []); // eslint-disable-line
 
     const changeParams = useCallback(action => {
```

Comment:
Here is the fix: since changeParams can be called several times in the same tick, we keep the result if the reducer locally so that we can start from it if another call comes immediately after. And we change the location (or the local state) only at the next tick, when all the sync changes were applied.